src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,238 @@
+            configItem.getService(),
+            existingCase.getId(),
+            exceptionRecord.id
+        );
+
+        try {
+            String s2sToken = s2sTokenGenerator.generate();
+
+            SuccessfulUpdateResponse updateResponse =
+                getUpdatedCaseFromService(exceptionRecord, configItem, existingCase, s2sToken);
+
+            log.info(
+                "Successfully called case update endpoint of service {} to update case with case Id {} "
+                    + "based on exception record ref {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+
+            if (!ignoreWarnings && !updateResponse.warnings.isEmpty()) {
+                // do not log warnings
+                return new ProcessResult(updateResponse.warnings, emptyList());
+            } else {
+                updateCaseInCcd(
+                    configItem.getService(),
+                    ignoreWarnings,
+                    idamToken,
+                    s2sToken,
+                    userId,
+                    exceptionRecord,
+                    updateResponse.caseDetails,
+                    existingCase
+                );
+
+                return new ProcessResult(
+                    exceptionRecordFinalizer.finalizeExceptionRecord(existingCase.getData(), existingCase.getId())
+                );
+            }
+        } catch (BadRequest exception) {
+            throw new CallbackException(
+                format(
+                    "Failed to call %s service Case Update API to update case with case Id %s "
+                        + "based on exception record %s",
+                    configItem.getService(),
+                    existingCase.getId(),
+                    exceptionRecord.id
+                ),
+                exception
+            );
+        } catch (UnprocessableEntity exception) {
+            ClientServiceErrorResponse errorResponse = serviceResponseParser.parseResponseBody(exception);
+            return new ProcessResult(errorResponse.warnings, errorResponse.errors);
+        } catch (Exception exception) {
+            throw new CallbackException(
+                format(
+                    "Failed to update case for %s service with case Id %s based on exception record %s",
+                    configItem.getService(),
+                    existingCase.getId(),
+                    exceptionRecord.id
+                ),
+                exception
+            );
+        }
+    }
+
+    private SuccessfulUpdateResponse getUpdatedCaseFromService(
+        ExceptionRecord exceptionRecord,
+        ServiceConfigItem configItem,
+        CaseDetails existingCase,
+        String s2sToken
+    ) throws CaseClientServiceException {
+        return caseUpdateClient.updateCase(
+            configItem.getUpdateUrl(),
+            existingCase,
+            exceptionRecord,
+            s2sToken
+        );
+    }
```

Comment:
don't need such a split - nothing else is happening and same arguments are passed

Reply:
Ok